src/interfaces/chain.cpp

```diff
@@ -306,10 +301,58 @@ class ChainImpl : public Chain
         auto it = ::mempool.GetIter(txid);
         return it && (*it)->GetCountWithDescendants() > 1;
     }
-    void relayTransaction(const uint256& txid) override
+    bool broadcastTransaction(const CTransactionRef& tx, const uint256& txid, CValidationState& state, const CAmount& absurdfee, bool relay) override
```

Comment:
Thanks for consolidating the code from `relayTransaction` and `submitToMemoryPool` in `interfaces/chain.cpp` and from `BroadcastTransaction` in `node/transaction.cpp`.

I do think the right place for all this code to live is in `node/transaction.cpp`, though. `interfaces/chain.cpp` is really just meant to be a simple glue file with lines like:

```c++
bool broadcastTransaction(...) override { return BroadcastTransaction(...); }
```

Also, I didn't look closely yet, but it seems like this refactoring might be changing wallet behavior (this could explain travis test failures). It would probably be better if this PR avoided changing behavior to the extent possible. This might require adding some extra options to the existing `BroadcastTransaction` function in `node/transaction.cpp`, or adding a second broadcast function there that can hopefully share some code with the current one.

Reply:
Okay was thinking than ```node/transaction.cpp``` was temporary and long term goal was to aggregate it to ```interfaces/node.cpp```.
Didn't want to change wallet behavior, fixed it by adding options in the existing BroadcastTransaction (but still not sure of ReacceptWalletTransactions, should we broadcast txn there ?), functional tests should be good now

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15713#discussion_r271524444

Thanks! This looks good and now net removes code (+63/-83).

> Okay was thinking than node/transaction.cpp was temporary

Yeah, that may have been the original thought. But I think it could be a reasonable long term location if more things are moved around it. In any case it's where the BroadcastTransaction function currently lives, so it's easier not to move in this PR, which is more about simplifying wallet code.

> and long term goal was to aggregate it to interfaces/node.cpp

Everything in interfaces/, including interfaces/node.cpp is just supposed to be very short glue code (the files would be huge otherwise). The interfaces are a way of letting wallet, node, and gui code run in different processes, and a way of preventing them from accessing each others internal state.